chain/chain/src/flat_storage_resharder.rs

```diff
@@ -2009,4 +2009,116 @@ mod tests {
     fn children_catchup_after_restart() {
         children_catchup_base(true);
     }
+
+    /// The split of a parent shard shouldn't happen until the resharding block has become final.
+    #[test]
+    #[ignore]
+    fn shard_split_should_wait_final_block() {
+        init_test_logger();
+        let (mut chain, resharder, sender) =
+            create_chain_resharder_sender::<DelayedSender>(simple_shard_layout());
+        let new_shard_layout = shard_layout_after_split();
+        let flat_store = resharder.runtime.store().flat_store();
+
+        // Add two blocks to the chain.
+        add_blocks_to_chain(&mut chain, 2);
+        assert_eq!(chain.head().unwrap().height, 2);
+        assert_eq!(chain.final_head().unwrap().height, 0);
+
+        // Trigger resharding at block 2 and it shouldn't split the parent shard.
+        let resharding_event_type = event_type_from_chain_and_layout(&chain, &new_shard_layout);
+        let ReshardingSplitShardParams { parent_shard, .. } = match resharding_event_type.clone() {
+            ReshardingEventType::SplitShard(params) => params,
+        };
+        assert!(resharder.start_resharding(resharding_event_type, &new_shard_layout).is_ok());
+        assert_eq!(sender.call_split_shard_task(), FlatStorageReshardingTaskStatus::Postponed);
+        assert!(flat_store.iter(parent_shard).count() > 0);
```

Comment:
nit: there is some rust library that has nice assert utilities like assert_gt, it may be a bit nicer, those usually provide better error messages than just doing comparison inside

Reply:
Cool, I'll switch to those better assertions!